src/storage/datasets_index.rs

```diff
@@ -43,206 +44,62 @@ impl DatasetsIndex {
+                });
+            };
         };
-        let dataset_1 = Arc::new(dataset_1);
-        let dataset_2 = Arc::new(dataset_2);
-        let exp_chunk_set: ChunkSet = [
-            ChunkRef {
-                dataset: dataset_1.clone(),
-                chunk: DataChunk {
-                    last_block: 697499.into(),
-                    first_block: 0.into(),
-                    last_hash: "f6275b81".into(),
-                    top: 0.into(),
-                },
-            },
-            ChunkRef {
-                dataset: dataset_1.clone(),
-                chunk: DataChunk {
-                    last_block: 962739.into(),
-                    first_block: 697500.into(),
-                    last_hash: "4eff9837".into(),
-                    top: 0.into(),
-                },
-            },
-            ChunkRef {
-                dataset: dataset_1.clone(),
-                chunk: DataChunk {
-                    last_block: 7308199.into(),
-                    first_block: 7300860.into(),
-                    last_hash: "6aeb1a56".into(),
-                    top: 7293060.into(),
-                },
-            },
-            ChunkRef {
-                dataset: dataset_2.clone(),
-                chunk: DataChunk {
-                    last_block: 192679.into(),
-                    first_block: 190280.into(),
-                    last_hash: "0999c6ce".into(),
-                    top: 0.into(),
-                },
-            },
-            ChunkRef {
-                dataset: dataset_2.clone(),
-                chunk: DataChunk {
-                    last_block: 3518379.into(),
-                    first_block: 3515280.into(),
-                    last_hash: "b8613f00".into(),
-                    top: 3510340.into(),
-                },
-            },
-        ]
-        .into_iter()
-        .collect();
+        chunk_set
+    }
 
-        let (chunk_set, ds_index) = parse_assignment(assignment).expect("Valid assignment");
-        assert_eq!(chunk_set, exp_chunk_set);
-        assert_eq!(ds_index.get_headers()["Auth"], "None");
-        assert_eq!(ds_index.list_files(&dataset_1, &chunk_set.first().unwrap().chunk), Some(vec![
-            super::RemoteFile {
-                url: Url::parse("https://example.com/0000000000/0000000000-0000697499-f6275b81/blocks.parquet").unwrap(),
-                name: "blocks.parquet".into(),
-            },
-            super::RemoteFile {
-                url: Url::parse("https://example.com/0000000000/0000000000-0000697499-f6275b81/transactions.parquet").unwrap(),
-                name: "transactions.parquet".into(),
-            },
-            super::RemoteFile {
-                url: Url::parse("https://example.com/0000000000/0000000000-0000697499-f6275b81/logs.parquet").unwrap(),
-                name: "logs.parquet".into(),
-            },
-        ]));
+    pub fn get_headers(&self) -> &reqwest::header::HeaderMap {
+        &self.http_headers
     }
-}
+}
```

Comment:
The file should always end with "\n". Please don't forget to run `cargo fmt`

Reply:
done